### PR TITLE
[SK] Add Ecto section and translate ecto/basics.md

### PR DESCRIPTION
**Pull request by gavlak**
- Add Ecto section in Slovak version
- Translate Ecto basics lesson

### CODE REVIEW

File: sk/lessons/ecto/basics.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+### Adaptéry
+
+Ecto podporuje rôzne databázy použitím adaptérov. Pár príkladov adaptérov sú:
+
+* PostgreSQL
+* MySQL
+* SQLite
+
+Pre túto lekciu nakonfigurujeme Ecto, aby používal PostgreSQL adaptér.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Pre túto lekciu nakonfigurujeme Ecto, aby používal adaptér PostgreSQL.
```

File: sk/lessons/ecto/basics.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+* Schémy — špeciálne štruktúry, ktoré reprezentujú záznamy v tabuľke
+
+Na začiatok si vytvoríme aplikáciu so supervision stromom.
+
+```shell
+$ mix new friends --sup
+$ cd friends
+```
+
+Pridáme ecto a postgrex balíčky ako závislosti do `mix.exs` súboru.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Do súboru `mix.exs` pridáme ako závislosti balíčky `ecto` a `postgrex`.
```

File: sk/lessons/ecto/basics.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+```elixir
+  defp deps do
+    [
+      {:ecto, "~> 2.0"},
+      {:postgrex, "~> 0.11"}
+    ]
+  end
+```
+
+Stiahneme závislosti použitím
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Stiahneme závislosti použitím príkazu
```

File: sk/lessons/ecto/basics.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+```shell
+$ mix deps.get
+```
+
+#### Vytvorenie Repozitára
+
+Repozitár v Ecte sa pripojí na dátové úložisko ako napríklad našu Postgres databázu.
+Všetka komunikácia s databázou bude vykonaná pomocou tohto repozitára.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Všetka komunikácia s databázou bude vykonávaná pomocou tohto repozitára.
```

File: sk/lessons/ecto/basics.md [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+Nastavíme repozitár spustením:
+
+```shell
+$ mix ecto.gen.repo -r Friends.Repo
+```
+
+Príkaz vygeneruje potrebnú konfiguráciu v `config/config.exs` na pripojenie k databáze vrátane adaptéru ktorý má použiť.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Príkaz vygeneruje potrebnú konfiguráciu v `config/config.exs` na pripojenie k databáze vrátane adaptéru, ktorý má použiť.
```

File: sk/lessons/ecto/basics.md [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+config :friends, Friends.Repo,
+  adapter: Ecto.Adapters.Postgres,
+  database: "friends_repo",
+  username: "postgres",
+  password: "",
+  hostname: "localhost"
+```
+
+Tu je nastavené ako sa Ecto pripojí na databázu.
+Všimnite si, ako je zvolený `Ecto.Adapters.Postgres` adaptér.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Všimnite si, že bol ako adaptér zvolený `Ecto.Adapters.Postgres`.
```

File: sk/lessons/ecto/basics.md [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+
+Tiež bol vytvorený `Friends.Repo` modul v `lib/friends/repo.ex`
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Tiež bol vytvorený modul `Friends.Repo` v súbore `lib/friends/repo.ex`
```

File: sk/lessons/ecto/basics.md [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+
+```elixir
+defmodule Friends.Repo do
+  use Ecto.Repo, otp_app: :friends
+end
+```
+
+Modul `Friends.Repo` budeme používať na vytvorenie dotazov na databázu. Tiež dodáme makru kľúč OTP aplikácie (`:friends`), podľa ktorého nájde informácie o konfigurácii.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Modul `Friends.Repo` budeme používať na dotazovanie databázy. Ako parameter dodáme kľúč, pod ktorým sa dá nájsť konfigurácia našej OTP aplikácie.
```

File: sk/lessons/ecto/basics.md [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+
+Ďalej nastavíme `Friends.Repo` ako supervízora v supervision strome našej aplikácie v `lib/friends/application.ex`.
+To nám zabezpečí, spustenie procesu Ecta, keď sa spustí naša aplikácia.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
To nám zabezpečí spustenie procesu Ecta, keď sa spustí naša aplikácia.
```

File: sk/lessons/ecto/basics.md [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+  def start(_type, _args) do
+    # List all child processes to be supervised
+    children = [
+      Friends.Repo,
+    ]
+
+  ...
+```
+
+Potom budeme musieť pridať nasledujúci riadok do nášho `config/config.exs` súboru:
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Potom budeme musieť do nášho súboru `config/config.exs` pridať nasledujúci riadok:
```

File: sk/lessons/ecto/basics.md [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+
+```elixir
+config :friends, ecto_repos: [Friends.Repo]
+```
+
+Toto umožní našej aplikácii spustiť ecto mix príkazy z príkazového riadku.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Toto umožní našej aplikácii spustiť `ecto mix` príkazy z príkazového riadku.
```

File: sk/lessons/ecto/basics.md [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+
+Práve sme úspešne nakonfigurovali repozitár!
+Môžeme vytvoriť Postgres databázu pomocou nasledujúceho príkazu:
+
+```shell
+$ mix ecto.create
+```
+
+Ecto použije informácie v `config/config.exs` súbore na určenie ako sa pripojiť do Postgresu a aké meno zvoliť pre databázu.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Ecto použije informácie v súbore `config/config.exs` na vytvorenie pripojenia do Postgresu a voľbu mena databázy.
```

File: sk/lessons/ecto/basics.md [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+
+Ak sa zobrazia nejaké chyby, overte, že nakonfigurované informácie sú správne a že inštancia Postgresu beží.
+
+### Migrácie
+
+Na vytváranie a modifikovanie tabuliek v postgres databáze nám Ecto poskytuje migrácie.
+Každá migrácia opisuje súbor akcií, ktoré majú byť vykonané na našej databáze, napr. ktoré tabuľky má vytvoriť alebo upraviť.
+
+Keďže naša databáza nemá zatiaľ žiadne tabuľky, vytvoríme migráciu a nejaké pridáme.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Keďže naša databáza nemá zatiaľ žiadne tabuľky, vytvoríme migráciu, ktorou nejaké pridáme.
```

File: sk/lessons/ecto/basics.md [hunk 14]
```diff
@@ -0,0 +1,230 @@
...
+Konvencia v Ecte je voliť názvy tabuliek v množnom čísle, čiže pre našu aplikáciu budeme potrebovať tabuľku `people` a tým začneme s našimi migráciami.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
V Ecte je konvenciou voliť názvy tabuliek v množnom čísle - pre našu aplikáciu budeme potrebovať tabuľku `people`.
```

File: sk/lessons/ecto/basics.md [hunk 15]
```diff
@@ -0,0 +1,230 @@
...
+
+Najlepší spôsob ako vytvoriť migrácie je mix príkaz `ecto.gen.migration <name>`, čiže v našom prípade:
+
+```shell
+$ mix ecto.gen.migration create_people
+```
+
+Toto nám vygeneruje nový súbor v zložke `priv/repo/migrations` a bude obsahovať timestamp v názve súboru.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Toto nám vygeneruje nový súbor v zložke `priv/repo/migrations`, ktorý obsahovať v názve súboru timestamp (časovú značku).
```

File: sk/lessons/ecto/basics.md [hunk 16]
```diff
@@ -0,0 +1,230 @@
...
+Keď tento súbor nájdeme v našom adresári a otvoríme migráciu mali by sme vidieť niečo podobné:
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Po otvorení vygenerovaního súboru (v adresári `priv/repo/migrations`) by sme mali vidieť niečo takéto:
```

**Comment by doomspork** [comment]
This is _awesome_ @gavlak! I'm going to give the @elixirschool/slovak team a chance to review, if they're unable to get to it, we'll go ahead and merge. 😁

File: sk/lessons/ecto/basics.md [hunk 17]
```diff
@@ -0,0 +1,230 @@
...
+  def change do
+    create table(:people) do
+      add :name, :string, null: false
+      add :age, :integer, default: 0
+    end
+  end
+end
+```
+
+Ako vidíme vyššie tiež sme definovali dátový typ stĺpca.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Ako vidíme vyššie, stĺpcom sme definovali dátový typ, obmedzenie na nenulovú hodnotu (`null: false`) a predvolenú hodnotu (`default: 0`).
```

File: sk/lessons/ecto/basics.md [hunk 18]
```diff
@@ -0,0 +1,230 @@
...
+Teraz ale poďme do príkazového riadku a spusťme našu migráciu:
+
+```shell
+$ mix ecto.migrate
+```
+
+### Schémy
+
+Potom ako sme vytvorili našú prvú tabuľku musíme Ectu o nej niečo povedať, čiastočne sa to robí pomocou použitia schém.
+Schéma je modul, ktorý definuje mapovanie na stĺpce danej tabuľky v databáze.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]: Mapovanie čoho na stĺpce tabuľky?

File: sk/lessons/ecto/basics.md [hunk 19]
```diff
@@ -0,0 +1,230 @@
...
+  use Ecto.Schema
+
+  schema "people" do
+    field :name, :string
+    field :age, :integer, default: 0
+  end
+end
+```
+
+Tu môžeme vidieť, že `Friends.Person` hovorí Ectu, že táto schéma popisuje tabuľku `people` a máme v nej dva stĺpce: `name`  ktorý je reťazec a `age`, celé číslo s predvolenou hodnotou `0`.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Tu môžeme vidieť, že `Friends.Person` hovorí Ectu, že táto schéma popisuje tabuľku `people` a máme v nej dva stĺpce: reťazec `name` a celé číslo`age` s predvolenou hodnotou `0`.
```

File: sk/lessons/ecto/basics.md [hunk 20]
```diff
@@ -0,0 +1,230 @@
...
+
+Poďme sa pozrieť ako vyzerá naša schéma pomocou príkazu `iex -S mix` a vytvorením novej osoby:
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Spustime si príkazom `iex -S mix` konzolu a pozrime sa, ako vyzerá naša schéma. Vyskúšajme si vytvoriť novú osobu:
```

File: sk/lessons/ecto/basics.md [hunk 21]
```diff
@@ -0,0 +1,230 @@
...
+
+```shell
+iex> %Friends.Person{}
+%Friends.Person{age: 0, name: nil}
+```
+
+Ako sme očakávali dostaneme štruktúru `Person` s predvolenou hodnotou pre atribút `age`.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Ako sme očakávali, dostaneme [štruktúru](https://elixirschool.com/sk/lessons/basics/modules/#structs) `Person` s predvolenou hodnotou pre atribút `age`.
```

File: sk/lessons/ecto/basics.md [hunk 22]
```diff
@@ -0,0 +1,230 @@
...
+"Tom"
+iex> Map.get(person, :name)
+"Tom"
+iex> %{name: name} = person
+%Friends.Person{age: 11, name: "Tom"}
+iex> name
+"Tom"
+```
+
+Podobne, môžeme upravovať naše schémy, ako by sme robili s akoukoľvek inou mapou alebo štruktúrou v Elixire:
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
Podobne, môžeme upravovať naše schémy, ako by sme robili s akoukoľvek inou [mapou](https://elixirschool.com/sk/lessons/basics/collections/#mapy) alebo [štruktúrou](https://elixirschool.com/sk/lessons/basics/modules/#structs) v Elixire:
```

File: sk/lessons/ecto/basics.md [hunk 23]
```diff
@@ -0,0 +1,230 @@
...
+
+```elixir
+iex> %{person | age: 18}
+%Friends.Person{age: 18, name: "Tom"}
+iex> Map.put(person, :name, "Jerry")
+%Friends.Person{age: 11, name: "Jerry"}
+```
+
+V ďalšej lekcii o Changesetoch sa pozrieme na to ako validovať zmeny našich dát a ako ich uložíme do našej databázy.
```
Comment (michalvalasek) [inline sk/lessons/ecto/basics.md]:
```suggestion
V ďalšej lekcii o Changesetoch sa pozrieme na to, ako validovať zmeny našich dát a ako ich uložiť do databázy.
```

**Comment by gavlak** [comment]
Appreciate the review @michalvalasek. All of the suggestions were great, I have incorporated them. 👍

**Comment by doomspork** [comment]
Thank you @gavlak and @michalvalasek :tada:

**Pull request merged**